### PR TITLE
Responsive header component

### LINKED ISSUE #225: UX Bug — Responsive header
**Job**
The header should be responsive to screen width

**Describe a possible solution**
Use collapse, toggles, screen size breakpoints to achieve responsiveness
Mainly collapse menu items on smaller screen sizes

**Context**
What it currently looks like on mobile:
Screen Shot 2020-10-10 at 2 55 14 PM

What is should look like on mobile:
Screen Shot 2020-10-10 at 2 53 36 PM

### PR DESCRIPTION
**Pull request by 123joshuawu**
**Issue**
Closes #225 

**Database Changes/Migrations**
N/A

**Test Modifications**
N/A

**Test Procedure**
1. Open site on mobile
2. Check if header looks cleannn (menu items collapsed), examples below

**Photos**
On mobile, header menu items are collapsed:
Screen Shot 2020-10-10 at 4 51 31 PM

On mobile, with menu items expanded:
Screen Shot 2020-10-10 at 4 51 39 PM

On desktop, no changes:
Screen Shot 2020-10-10 at 4 51 50 PM

**Additional Info**
The dark mode toggle is a little off, broke that off to #228

### CODE REVIEW

File: src/web/src/components/Header.vue [hunk 1]
```diff
@@ -1,109 +1,87 @@
...
+              Explore
+            </router-link>
+          </b-nav-item>
+        </b-navbar-nav>
+        <!-- If user has logged in -->
+        <b-navbar-nav class="ml-auto">
+          <b-nav-item-dropdown right v-if="sessionID !== null">
+            <!-- Using 'button-content' slot -->
+            <template v-slot:button-content>Hi, {{ userName }}</template>
+            <!-- <b-dropdown-item href="#">Profile</b-dropdown-item> -->
```
Comment (jshom) [inline src/web/src/components/Header.vue:27]: Please remove unused code comments, lines 25,27,33.
Besides that lgtm!

**Review by jshom — CHANGES_REQUESTED** [review]
Sweeet!

Maybe we change the bg color of one of the buttons here, feels a bit too close.
Also, would prefer to handle the offset darkmode toggle in this issue.

Thanks!

Screen Shot 2020-10-11 at 11 03 48 PM

Once these are fixed, should be good to merge

File: src/web/src/components/Header.vue (hunk 1 above)
Comment (123joshuawu) [reply src/web/src/components/Header.vue]: Removed

File: src/web/src/components/Login.vue [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+          );
+        });
+      this.$emit("submit");
+      // this.$refs["login-modal"].hide();
+    },
+    onReset(evt) {
+      evt.preventDefault();
+      // Reset our form values
+      this.form.email = "aaa1@wa.com";
+      this.form.password = "123456";
```
Comment (jshom) [inline src/web/src/components/Login.vue:73]: Lets set these to empty strings, http://wa.com/ isn't gonna like it tho :)

**Review by jshom — APPROVED** [review]
Few small tweaks, and ready to merge. LOOOOKS AWESOMEE!!! 🎊

File: src/web/src/components/Login.vue [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            `Login Unsuccesful. Please double check your email and password, then try again!`,
+            {
+              title: "Invalid login",
+              variant: "danger",
+              noAutoHide: true,
+            }
+          );
+        });
+      this.$emit("submit");
+      // this.$refs["login-modal"].hide();
```
[the remark this answers is not in the document]
Comment (123joshuawu) [reply src/web/src/components/Login.vue:67]: Removed

File: src/web/src/components/Login.vue [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+      login(userInfo)
+        .then((response) => {
+          console.log(response);
+          this.$cookies.set("sessionID", response.data.content["sessionID"]);
+          this.$cookies.set("userName", response.data.content["userName"]);
+          this.$cookies.set("userID", response.data.content["uid"]);
+          location.reload();
+        })
+        .catch((error) => {
+          console.log(error.response);
```
[the remark this answers is not in the document]
Comment (123joshuawu) [reply src/web/src/components/Login.vue:56]: True, changed

File: src/web/src/components/Login.vue (hunk 2 above)
Comment (123joshuawu) [reply src/web/src/components/Login.vue:73]: lool yuze be like "wa"

File: src/web/src/assets/_bootstrap_helpers.scss [hunk 5]
```diff
@@ -0,0 +1,5 @@
+// Do not edit or add anything here
```
[the remark this answers is not in the document]
Comment (123joshuawu) [reply src/web/src/assets/_bootstrap_helpers.scss:1]: yeah, added more context

**Pull request merged**